airbyte-webapp/src/components/base/Text/Text.tsx

```diff
@@ -0,0 +1,68 @@
+import classNames from "classnames";
+import React from "react";
+
+import headingStyles from "./heading.module.scss";
+import textStyles from "./text.module.scss";
+
+type TextSize = "sm" | "md" | "lg";
+type HeadingSize = TextSize | "xl";
+type TextElementType = "p" | "span" | "div";
+type HeadingElementType = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
+
+interface BaseProps {
+  className?: string;
+  centered?: boolean;
+}
+
+interface TextProps extends BaseProps {
+  as?: TextElementType;
+  size?: TextSize;
+  bold?: boolean;
+}
+
+interface HeadingProps extends BaseProps {
+  as?: HeadingElementType;
+  size?: HeadingSize;
+}
+
+const getTextClassNames = ({ size, centered, bold }: Required<Pick<TextProps, "size" | "centered" | "bold">>) => {
```

Comment:
```
Required<Pick<TextProps, "size" | "centered" | "bold">>
```
To confirm my understanding - the purpose of this is basically to remove the `as` and `className` fields from `TextProps`, because we don't actually want those to be class names on the component that we are returning below, correct?

If so, would it potentially be better to do something like `Required<Omit<TextProps, "className" | "as">>` instead, so that if we add more classname props to those types in the future we don't need to also edit these function signatures? Probably not a huge deal either way

Reply:
You'd need to add the new property to the destructure anyway.
No matter what method, there's type safety if size, centered, or bold are changed in TextProps.

Reply:
In case that the props were to expand, this would continue to work with Pick over using Omit.

Reply:
Ah, good point that the new props would need to be added to the destructure, I forgot about that. This looks good to me then!